bpfd/src/storage.rs

```diff
@@ -100,21 +116,147 @@ impl Node for CsiNode {
         let volume_id = &req.volume_id;
         let target_path = &req.target_path;
         let volume_context = &req.volume_context;
+        // TODO (astoycos) support readonly bpf pins.
         let read_only = &req.readonly;
 
         info!(
             "Received publish volume request with :\n\
-        volume_caps: {volume_cap:#?},\n\
-        volume_id: {volume_id},\n\
-        target_path: {target_path},\n\
-        volume_context: {volume_context:#?},\n\
-        read_only: {read_only}\n\
-        Sleeping for 60 seconds"
+                volume_caps: {volume_cap:#?},\n\
+                volume_id: {volume_id},\n\
+                target_path: {target_path},\n\
+                volume_context: {volume_context:#?},\n\
+                read_only: {read_only}"
         );
 
-        tokio::time::sleep(std::time::Duration::from_secs(60)).await;
+        match (
+            volume_context.get(MAPS_KEY),
+            volume_context.get(PROGRAM_KEY),
+        ) {
+            (Some(m), Some(program_name)) => {
+                let maps: Vec<&str> = m.split(',').collect();
+
+                // Get the program information from the BpfManager
+                let (resp_tx, resp_rx) = oneshot::channel();
+                let cmd = Command::List { responder: resp_tx };
+
+                let tx = self.tx.lock().unwrap().clone();
+                // Send the LIST request
+                tx.send(cmd).await.unwrap();
+
+                // Await the response
+                let core_map_path = match resp_rx.await {
+                    Ok(res) => match res {
+                        // Find the Program with the specified *Program CRD name
+                        Ok(results) => {
+                            let prog_data = results
+                                .into_iter()
+                                .flat_map(|p| p.take_data())
+                                .find(|d| {
+                                    d.metadata().get(OPERATOR_PROGRAM_KEY) == Some(program_name)
+                                })
+                                .ok_or(Status::new(
+                                    5.into(),
```

Comment:
What is this magic number and why is it a status?

Reply:
:) https://github.com/container-storage-interface/spec/blob/master/spec.md#nodepublishvolume-errors

I'll make these constants